### PR TITLE
optim 剩余部分

### PR DESCRIPTION
**Pull request by robin978**
[oneflow.optim.RMSprop](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/rmsprop.py#L24)
[oneflow.optim.SGD](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/sgd.py#L25)
截屏2022-03-24 下午3 09 20
截屏2022-03-24 下午3 09 28
截屏2022-03-24 下午3 09 36

截屏2022-03-24 下午3 09 43

[oneflow.optim.CosineAnnealingLR](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/cosine_annealing_lr.py#L22)
[oneflow.optim.CosineDecayLR](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/cosine_decay_lr.py#L22)

截屏2022-03-24 下午3 10 01

截屏2022-03-24 下午3 10 13
截屏2022-03-24 下午3 10 20

### CODE REVIEW

**Comment by robin978** [comment]
[oneflow.optim. ExponentialLR](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/exponential_lr.py#L20)
截屏2022-03-24 下午3 10 26

[oneflow.optim.LambdaLR](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/lambda_lr.py#L21)
截屏2022-03-24 下午3 10 31

[oneflow.optim. MultiStepLR](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/multistep_lr.py#L22)
截屏2022-03-24 下午3 10 39

[oneflow.optim. PolynomialLR](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/polynomial_lr.py#L22)
截屏2022-03-24 下午3 10 46

**Comment by robin978** [comment]
[oneflow.optim. ReduceLROnPlateau](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/reduce_lr_on_plateau.py#L20)
截屏2022-03-28 下午1 54 34
截屏2022-03-28 下午1 54 54

[oneflow.optim.steplr](https://github.com/Oneflow-Inc/oneflow/blob/c5f52665655e8a1cdac1a0c309dec7af9add7bad/python/oneflow/nn/optimizer/step_lr.py#L22)
截屏2022-03-28 下午1 55 01

File: docs/source/cn/lambda_lr.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    将每个参数组的学习率设置为初始 lr 乘一个给定的函数。
+    当 last_epoch=- 1时，设置初始学习率为 lr。
+
+    .. math::
+
+        learning\\_rate = base\\_learning\\_rate*lambda(last\\_step)
+
+    参数：
+        - **optimizer** (Optimizer) - 封装的优化器。
+        - **lr_lambda** (function or list) - 一个给定整数参数 epoch 的计算乘法因子的函数，
```
Comment (rainyq69) [inline docs/source/cn/lambda_lr.py:16]:
```suggestion
        - **lr_lambda** (function or list) - 一个根据给定参数 epoch 计算乘法因子的函数，或为一组函数，其中每个函数用于 optimizer.param_groups 中的每一个组。
```

File: docs/source/cn/lambda_lr.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        参数：
+            - **state_dict** (dict) - 调整器的状态，应为调用 :meth:`state_dict` 函数所返回的对象。
+        """
+)
+
+reset_docstr(
+    oneflow.optim.lr_scheduler.LambdaLR.state_dict,
+    """ 以 :class:`dict` 形式返回调整器的状态。
+
+        它包含了 self.__dict__ 中每个变量的条目，而这些变量并不是优化器。
```
Comment (rainyq69) [inline docs/source/cn/lambda_lr.py:51]:
```suggestion
        它包含了 self.__dict__ 中每个不是优化器的变量的条目。
```

File: docs/source/cn/polynomial_lr.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    .. math::
+        \begin{aligned}
+           & decay\_batch = min(decay\_batch, current\_batch) \\
+           & learning\_rate = (base\_lr-end\_lr)*(1-\frac{current\_batch}{decay\_batch})^{power}+end\_lr
+        \end{aligned}
+
+    参数:
+        - **optimizer** (Optimizer) - 封装的优化器。
+        - **steps** (int) - decayed steps。
```
Comment (rainyq69) [inline docs/source/cn/polynomial_lr.py:27]: 未翻译。

File: docs/source/cn/reduce_lr_on_plateau.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from docreset import reset_docstr
+
+reset_docstr(
+    oneflow.optim.lr_scheduler.ReduceLROnPlateau,
+    """当一个 metric 停止更新时，降低学习率。
+    一旦学习停滞不前，模型通常会从降低2-10倍的学习率中受益。
+    这个调整器读取一个指标量，如果在 'patience' 的 epoch 数中没有看到改进，
+    学习率就会降低。
+
+    参数:
```
Comment (rainyq69) [inline docs/source/cn/reduce_lr_on_plateau.py:11]: 这几个参数里的换行记得都删掉

File: docs/source/cn/step_lr.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import oneflow
+from docreset import reset_docstr
+
+reset_docstr(
+    oneflow.optim.lr_scheduler.StepLR,
+    """
+    在每一个 step_size 大小的步骤中，以伽玛衰减每个参数组的学习率。
+    注意，这种衰减可以与本调整器以外的其他学习率变化同时发生。
```
Comment (rainyq69) [inline docs/source/cn/step_lr.py:8]:
```suggestion
    注意，这种衰减可以与本调整器无关的其他的学习率变化同时发生。
```

File: docs/source/cn/cosine_annealing_lr.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    当 last_step=-1， 设置初始学习率为 lr。因为时间表是递归定义的， 在这个学习率调整器之外，它也可以同时被其他操作者修改。
+    如果学习率完全由这个调整器设定，每一步的学习率就变成了：
+
+    .. math::
+        \eta_t = \eta_{min} + \frac{1}{2}(\eta_{max} - \eta_{min})\left(1 +
+        \cos\left(\frac{T_{cur}}{T_{max}}\pi\right)\right)
+
+    它在 `SGDR: Stochastic Gradient Descent with Warm Restarts`_ 中被提出。
+    请注意，这只实现了 SGDR 的 cosine annealing 部分，而不是重新启动。
```
Comment (robin978) [inline docs/source/cn/cosine_annealing_lr.py:30]:
```suggestion
    请注意，这只实现了 SGDR 的 cosine annealing 部分，并不包括重新启动。
```

File: docs/source/cn/cosine_annealing_lr.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    参数:
+        - **optimizer** (Optimizer) - 封装的优化器。
+        - **T_max** (int) - 迭代的最大数量。
+        - **eta_min** (float) - 学习率的最小值，默认值：0。
+        - **last_step** (int) - 最后一个 epoch 的索引，默认值：-1。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout，默认值: ``False`` 。
```
Comment (robin978) [inline docs/source/cn/cosine_annealing_lr.py:37]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到标准输出，默认值: ``False`` 。
```

File: docs/source/cn/cosine_decay_lr.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+    .. math::
+
+        learning\\_rate = {base\\_learning\\_rate}*{\\alpha}
+
+    它在 `SGDR: Stochastic Gradient Descent with Warm Restarts`_ 中被提出。
+    请注意，这里只实现了 SGDR 的 cosine annealing 部分，而不是重新启动。
+
+    参数：
+        - **optimizer** (Optimizer) - 封装的优化器。
```
Comment (robin978) [inline docs/source/cn/cosine_decay_lr.py:28]:
```suggestion
        - **optimizer** (Optimizer) - 被包装的优化器。
```

File: docs/source/cn/exponential_lr.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+from docreset import reset_docstr
+
+reset_docstr(
+    oneflow.optim.lr_scheduler.ExponentialLR,
+    """
+    在每个 epoch 中以 gamma 衰减每个参数组的学习率。
+    当 last_epoch=- 1时，设置初始学习率为 lr。
+
+    参数：
+        - **optimizer** (Optimizer) - 封装的优化器。
```
Comment (robin978) [inline docs/source/cn/exponential_lr.py:11]:
```suggestion
        - **optimizer** (Optimizer) - 被包装的优化器。
```

File: docs/source/cn/exponential_lr.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+        - **gamma** (float) - 学习率衰减的乘法系数。
+        - **last_step** (int) - 最后一个 epoch 的索引，默认值：-1。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout，默认值: ``False`` 。
```
Comment (robin978) [inline docs/source/cn/exponential_lr.py:14]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到标准输出，默认值: ``False`` 。
```

File: docs/source/cn/lambda_lr.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    将每个参数组的学习率设置为初始 lr 乘一个给定的函数。
+    当 last_epoch=- 1时，设置初始学习率为 lr。
+
+    .. math::
+
+        learning\\_rate = base\\_learning\\_rate*lambda(last\\_step)
+
+    参数：
+        - **optimizer** (Optimizer) - 封装的优化器。
```
Comment (robin978) [inline docs/source/cn/lambda_lr.py:15]:
```suggestion
        - **optimizer** (Optimizer) - 被包装的优化器。
```

File: docs/source/cn/multistep_lr.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+from docreset import reset_docstr
+
+reset_docstr(
+    oneflow.optim.lr_scheduler.MultiStepLR,
+    """
+    一旦步数达到一个临界值，每个参数组的学习率就会以 gamma 衰减。请注意，这种衰减可能与来自本调整器之外的其他学习率变化同时发生。
+    当 last_epoch=- 1时，设置初始学习率为 lr。
+
+    参数：
+        - **optimizer** (Optimizer) - 封装的优化器。
```
Comment (robin978) [inline docs/source/cn/multistep_lr.py:11]:
```suggestion
        - **optimizer** (Optimizer) - 被包装的优化器。
```

File: docs/source/cn/multistep_lr.py [hunk 13]
```diff
@@ -0,0 +1,32 @@
+import oneflow
+from docreset import reset_docstr
+
+reset_docstr(
+    oneflow.optim.lr_scheduler.MultiStepLR,
+    """
+    一旦步数达到一个临界值，每个参数组的学习率就会以 gamma 衰减。请注意，这种衰减可能与来自本调整器之外的其他学习率变化同时发生。
+    当 last_epoch=- 1时，设置初始学习率为 lr。
```
Comment (robin978) [inline docs/source/cn/multistep_lr.py:8]:
```suggestion
    当 last_epoch =- 1时，设置初始学习率为 lr。
```

File: docs/source/cn/multistep_lr.py [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+    参数：
+        - **optimizer** (Optimizer) - 封装的优化器。
+        - **milestones** (list) - step indices 的 list，必须是增加的。
+        - **gamma** (float, optional) - 学习率衰减的乘法系数（默认值：0.1）。
+        - **last_step** (int) - 最后一个 epoch 的索引（默认值：-1）。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout（默认值: ``False`` ）。
```
Comment (robin978) [inline docs/source/cn/multistep_lr.py:15]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到标准输出（默认值: ``False`` ）。
```

File: docs/source/cn/polynomial_lr.py [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    如果 cycle 为 `False` ，则等式为：
+
+    .. math::
+        \begin{aligned}
+           & decay\_batch = min(decay\_batch, current\_batch) \\
+           & learning\_rate = (base\_lr-end\_lr)*(1-\frac{current\_batch}{decay\_batch})^{power}+end\_lr
+        \end{aligned}
+
+    参数:
+        - **optimizer** (Optimizer) - 封装的优化器。
```
Comment (robin978) [inline docs/source/cn/polynomial_lr.py:26]:
```suggestion
        - **optimizer** (Optimizer) - 被包装的优化器。
```

File: docs/source/cn/reduce_lr_on_plateau.py [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+        - **threshold_mode** (str) - `rel` 或 `abs` 。 在 `rel` 模式中，
+            'max' ：dynamic_threshold = best * ( 1 + threshold )，
+            'min' ：dynamic_threshold = best * ( 1 - threshold )； 
+            在 `abs` 模式中，`max` ：dynamic_threshold = best + threshold，
+            `min` ：dynamic_threshold = best - threshold inmode。默认： `rel` 。
+        - **cooldown** (int) - 学习率减少后，在恢复正常操作前要等待的 epoch 数，默认值：0。
+        - **min_lr** (float or list) - 一个标量或一个标量列表，
+            分别是所有参数组或每组的学习率下限，默认：0。
+        - **eps** (float) - 适用于学习率的最小衰减，如果新旧学习率之间的差异小于 eps，更新将被忽略。默认值：1e-8。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout，默认值: ``False`` 。
```
Comment (robin978) [inline docs/source/cn/reduce_lr_on_plateau.py:30]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 标准输出，默认值: ``False`` 。
```

File: docs/source/cn/rmsprop.py [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+            # 读取数据，计算损失，等等。
+            # ...
+            loss.backward()
+            rmsprop.clip_grad()
+            rmsprop.step()
+            rmsprop.zero_grad()
+
+    若要使用 clip_grad 函数，请参考这个示例。
+
+    关于 `clip_grad_max_norm` 和 `clip_grad_norm_type` 函数的更多细节，请参考 :func:`oneflow.nn.utils.clip_grad_norm` 。
```
Comment (robin978) [inline docs/source/cn/rmsprop.py:99]:
```suggestion
    关于 `clip_grad_max_norm_` 和 `clip_grad_norm_type` 函数的更多细节，请参考 :func:`oneflow.nn.utils.clip_grad_norm` 。
```

File: docs/source/cn/step_lr.py [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+    当 last_step = -1 时，设置初始学习率为 lr。
+
+    参数:
+        - **optimizer** (Optimizer) - 封装的优化器。
```
Comment (robin978) [inline docs/source/cn/step_lr.py:12]:
```suggestion
        - **optimizer** (Optimizer) - 被包装的优化器。
```

File: docs/source/cn/step_lr.py [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+        - **step_size** (int) - 学习率衰减的时间。
+        - **gamma** (float, optional) - 学习率衰减的乘法系数（默认：0.1）。
+        - **last_step** (int) - 最后一个 epoch 的索引（默认值：-1）。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout（默认值: ``False`` ）。
```
Comment (robin978) [inline docs/source/cn/step_lr.py:16]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到标准输出（默认值: ``False`` ）。
```

File: docs/source/cn/cosine_decay_lr.py [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+
+    它在 `SGDR: Stochastic Gradient Descent with Warm Restarts`_ 中被提出。
+    请注意，这里只实现了 SGDR 的 cosine annealing 部分，而不是重新启动。
+
+    参数：
+        - **optimizer** (Optimizer) - 被包装的优化器。
+        - **decay_steps** (int) - 学习率调整器中 decay steps。
+        - **alpha** (float, optional) - 学习率比例因子（ :math:`\\alpha` ）（默认值： 0.0）。
+        - **last_step** (int) - 最后一个 epoch 的索引，默认值：-1。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout，默认值: ``False`` 。
```
Comment (robin978) [inline docs/source/cn/cosine_decay_lr.py:32]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到标准输出，默认值: ``False`` 。
```

File: docs/source/cn/lambda_lr.py [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+    .. math::
+
+        learning\\_rate = base\\_learning\\_rate*lambda(last\\_step)
+
+    参数：
+        - **optimizer** (Optimizer) - 被包装的优化器。
+        - **lr_lambda** (function or list) - 一个根据给定参数 epoch 计算乘法因子的函数，或为一组函数，其中每个函数用于 optimizer.param_groups 中的每一个组。
+            或者一个此类函数的列表，一个在每组 optimizer.param_groups 中。
+        - **last_step** (int) - 最后一个 epoch 的索引（默认值：-1）。
+        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到 stdout（默认值: ``False`` ）。
```
Comment (robin978) [inline docs/source/cn/lambda_lr.py:19]:
```suggestion
        - **verbose** (bool) - 如果为 ``True`` ，则会为每次更新打印一条信息到标准输出（默认值: ``False`` ）。
```

**Review by rainyq69 — APPROVED** [review]

**Pull request merged**